frontend/public/components/marketplace/kubernetes-marketplace.jsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+
+import { Firehose, PageHeading, StatusBox } from '../utils';
+import { referenceForModel } from '../../module/k8s';
+import { normalizeIconClass } from '../catalog-item-icon';
+import { PackageManifestModel } from '../../models';
+import CatalogTileView from 'patternfly-react-extensions/dist/esm/components/CatalogTileView/CatalogTileView';
+import CatalogTile from 'patternfly-react-extensions/dist/esm/components/CatalogTile/CatalogTile';
+
+class MarketplaceListPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Please see https://github.com/openshift/console/pull/612, `componentDidUpdate` is not called on initial render, so items may not initially load correctly

Reply:
Actually, I think it'd be better to use `getDerivedStateFromProps` instead of `componentDidUpdate`

Reply:
`getDerivedStateFromProps()` is considered legacy and you should avoid it.

Reply:
um... don't think so. where did you hear that?

Reply:
https://reactjs.org/docs/react-component.html but I'm think I misread which ones are legacy, their documentation could be better layed out :-)

Reply:
```So Dan Abramov answered on Twitter and it seems like there are 2 reasons why you should use getDerivedStateFromProps instead of componentDidUpdate + setState:

setState in componentDidUpdate causes an extra render (not directly perceptible to user but it slows down your app). And you render method can't assume the state is ready (because it won't be the first time).

Performances reason: it avoids unnecessary re-render.
As getDerivedStateFromProps is called before rendering on init, you can initialise your state in this function instead of having a constructor to do so. Currently you had to have a constructor or componentWillMount to init your state before initial rendering.```

That would explain your need to set state in the constructor.

Reply:
@dtaylor113 just pushed this change - let me know if it looks good to you